client/components/command-palette/use-capabilities-query.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+export const useCapabilitiesQuery = () =>
+	useQuery( {
+		queryKey: [ 'command-palette', 'capabilities' ],
+		queryFn: () => {
+			return request( {
+				path: '/me/sites',
+				apiVersion: '1.2',
+				query: new URLSearchParams( {
+					fields: 'ID,capabilities',
+					site_visibility: 'all',
+					site_activity: 'active',
+					include_domain_only: 'true',
+				} ).toString(),
+			} );
+		},
+		select: ( data ) => {
+			const capabilities = {};
+			data.sites.forEach( ( site ) => {
+				capabilities[ site.ID ] = site.capabilities;
+			} );
+			return capabilities;
```

Comment:
I think we need to type the return value. I'm seeing some type errors on `use-command-palette`  :191. And some more errors at other places.

Reply:
Yeah, I didn't pay attention to types yet. I wanted to build something functional first.